### ISSUE TITLE
LVM doesn't work

### ISSUE THREAD

**Issue by maverick85** [Antergos/Cnchi#1121]
**Describe the bug**
Automatic installation with LVM doesn't boot

**To Reproduce**
1. Burn ISO (antergos-minimal-19.1-x86_64) to USB
2. Boot from USB
3. Read below

**Expected behavior**
A working installation

**Screenshots**

**Additional context**
Ok, so I am used to red hat based distros, but I got an old Sony Vaio and I thought of installing Antergos on it to see how it goes.
After burning the ISO to USB and booting, I faced several issues:

Wi-Fi fails - when I select the appropriate network, it fails to connect. It's protected using WPA PSK, it never asks me for the password. Instead I have to select: "Join an Hidden Network" and insert everything manually. wtf! but ok...

... proceed with it.

Select your location:
My country, territory or area is PORTUGAL. But all my computing devices are in english, it's the language I use across the board.
But here you read "Please, note that your system language will be determined from this selection"
Since I haven't made it to successfully boot the system after install, I'll just take your word for this retarded configuration if it comes to fruition.

If you happen to have an AMD gfx card, don't you make the error of selecting proprietary drivers. Instead of saying those are not avail, and proceed the installation with Linux drivers, it simply breaks the installation saying catalyst drivers weren't found.

'Cannot find these packages: catalyst-utils, catalyst-hook, lib32-catalyst-utils, lib32-catalyst-libgl, lib32-opencl-catalyst, catalyst-libel': add_fatal in /usr/share/chchi/src/misc/events.py:83

Wash, rinse and repeat.

Here we go again, feature selection, without proprietary drivers. Let's move forward.
I was coming through this and upon the installation I kept getting errors on install kept failing. All was happing on the phase of "Checking that all selected packages are available online".

'Error creating metalling for package xorg-xinput. Installation will stop': add_fatal in /usr/share/chchi/src/misc/events.py:83

I actually was starting to think it could have something to do with my wifi card or whatever. So I connected using ethernet... Same issue. Thinking about the "Checking that all selected packages are available online". ... and I remembered that I always used the option "Let Cnchi sort the mirrors list (recommended)".
So, I selected "Manage the mirrors lists manually (advanced)" and selected most of the Antergos mirrors.

VOILA, no more errors on "Checking that packages are available" what 19.1 version is this that the installer can't properly manage the mirrors? I am truly appalled.
But yeah, sure. Installation is advancing, will I finally be able to try this out?

NO !

ERROR:

Antergos Installer - Confirmation
IMPORTANT: There may have been a problem with the boot loader installation which could prevent your system from booting properly. Before rebooting, you may want to verify whether or not the boot loader is installed and configured.

The Arch Linux Wiki contains troubleshooting information: https://wiki.archlinux.org/indexphp/GRUB

Would you like to view the wiki page now?

No/Yes"

Well, nevermind answering no or yes, none will open any page, instead will simply invite you to reboot the system.

After rebooting, I am taken to the `grub>` prompt
...

I cannot express how cheap and poorly built this looks. It feels like shitbuntu, and works even worse.
I came down to understand that it is a limitation of Arch+LVM.

Yes, Logical Volume Manager! Something which is pretty much standard, should be the default option, yet, Cnchi has no clue on how to properly install to boot. The installer doesn't allow to properly configure LVM during setup, and it's a major headache to come around and put it to work.

If you select the option to manually configure partitioning, it doesn't have options that allow to properly configure LVM either.

And since I want an OS that allows me to move through it, not to stop at every step of the way, I'm abandoning this. I could go and configure everything manually, but that's not what I am looking for to do with my free time right now. I have seen two, three old threads about this, there has been more than time to make a installer that gets all the way through.

Hope the above post gets someone working better, if not... not my loss. Have a good one!

![antergos - bootloader](https://user-images.githubusercontent.com/10200278/51044956-4bb89880-15ba-11e9-927d-c57bb2786c17.jpg)
![antergos-catalyst](https://user-images.githubusercontent.com/10200278/51044957-4c512f00-15ba-11e9-8328-87e8e1ce1c97.jpg)
![antergos-xorg-xinput](https://user-images.githubusercontent.com/10200278/51044958-4c512f00-15ba-11e9-9bd2-8a38889137d7.jpg)

**Comment by killajoe** [comment]
i do a try install with xfce + encryption + lvm (choosing defaults) and it seems to install grub2-zfs ?
[cnchi.log](https://github.com/Antergos/Cnchi/files/2751958/cnchi.log)

**Comment by maverick85** [comment]
I actually ditched this for a Mint installation got it up and running in 20.
Aside from the issues mentioned above, there are actually a few others I didn't mention, like:

When the ISO boots, there's an option to 'Boot from Harddrive' ... does nothing comes back to the exact same screen;
If you select the option for 'non-graphical boot' takes you to the exact same graphical boot as the first option.

Apart from that, I've spent a couple of hours trying to fix this.
I have also tried other things, like:

After failed boot went on the LiveISO and mounted the LVM AntergosRoot LV.
Since my boot partition was sda1 and the LVM Volume was sda2, I also mounted sda1 to /mnt/boot
I also mounted /run to /mnt/hostrun w bind and then in chroot linked hostrun to run.

Then I went on with grub-install (target=i386-pc) and grub-mkconfig.
All went through with no errors, then upon reboot I got back to the `grub>` prompt, although the grub.cfg being inside sda1 boot/grub/grub.cfg (with lvm support etc I checked it).

Anyway, Mint found no issues, I just wanted to use some old ass hardware to stream tv. Done.

**Comment by killajoe** [comment]
https://wiki.archlinux.org/index.php/GRUB#Device_/dev/xxx_not_initialized_in_udev_database_even_after_waiting_10000000_microseconds

Is the issue... i do install Antergos with LVM two times and it do not generate grub.cfg in both installs

**Comment by killajoe** [comment]
> Good luck in getting an answer from the community or any developer though. Seen cemeteries with more movement.

yea i do not know what happens to Antergos devs... there is no commit or communication for 10 days now...

**Comment by maverick85** [comment]
Yes, unfortunately. Anyway as I mentioned earlier I was just looking for an easy install for an old Sony Vaio laptop so I could stream some TV to it and connect it to a TV. I wasn't expecting to spend more than 30 minutes on the whole process, and I spent a day around trying to figure all the errors I came across.

Anyway, if you may please try this and let me know if it works, after you get booted to the grub shell:
```
$ insmod lvm
$ ls
(lvm/AntergosVG-AntergosSwap) (lvm/AntergosVG-AntergosRoot) (hd0) (hd0,msdos2) (hd0,msdos1)
$ set root=(hd0, msdos1)
$ linux /vmlinuz-linux root=/dev/mapper/AntergosVG-AntergosRoot
$ initrd /initramfs-linux.img
$ boot
```
Then after boot run this command `grub-mkconfig -o /boot/grub/grub.cfg`

Cheers

**Comment by jiggahertz** [comment]
A similar issue appeared in December's Arch image. `grub-mkconfig -o /boot/grub/grub.cfg` would hang on a blank prompt without any output. The following commands fixed the issue:

```
mkdir /mnt/hostlvm
mount --bind /run/lvm /mnt/hostlvm
arch-chroot /mnt
ln -s /hostlvm /run/lvm
```

Then run `grub-mkconfig -o /boot/grub/grub.cfg`and proceed as normal. Hopefully the devs fix the installer soon, but I think they've kinda surrendered to Manjaro at this point.
